app/models/ability.rb

```diff
@@ -0,0 +1,16 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    user ||= User.new # Guest user (not logged in)
+
+    if user.admin?
+      can :manage, :all # Admin has full access to all resources
+    else
+      can :read, Post # Regular users can read posts
```

Comment:
Is keeping `Post` here necessary?

Reply:
I have fixed it. Thanks for pointing it out.